libraries/provider_configure.rb

```diff
@@ -0,0 +1,91 @@
+class Chef
+  # Chef Provider for configuring an elasticsearch instance
+  class Provider::ElasticsearchConfigure < Provider
+    include Poise
+
+    def action_manage
+      # Create ES directories
+      #
+      [ new_resource.path_conf, new_resource.path_logs ].each do |path|
+        directory path do
+          owner new_resource.user
+          group new_resource.group
+          mode 0755
+          recursive true
+          action :create
+        end
+      end
+
+      # Create data path directories
+      #
+      data_paths = new_resource.path_data.is_a?(Array) ? new_resource.path_data : new_resource.path_data.split(',')
+
```

Comment:
What if we split these different features into separate Ruby methods, and then call them in the `action_manage` method, in a "composed method" pattern? Is it worthwile to do it? Would that be an antipattern in Chef? Would that make testing more transparent (ie. we can test separately thate `create_data_path_directories` creates the folders, and analogically with `create_user`, `create_logging_configuration_file`)?